hudi-hive/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -163,6 +163,11 @@ private void syncSchema(boolean tableExists, boolean isRealTime, MessageType sch
    */
   private void syncPartitions(List<String> writtenPartitionsSince) {
     try {
+      if (cfg.useHiveStylePartitioning) {
+        LOG.info("Sync partitions through MSCK Repair");
+        hoodieHiveClient.syncPartitionsByMSCK();
```

Comment:
@zhedoubushishi : Wouldn't "msck repair" list all partitions in DFS (S3). With commit metadata, we can identify the list of partitions without listing all data partitions. Can we leverage this here ?

Reply:
I see. Then it seems we should not use "MSCK" here.